skorch/callbacks/training.py

```
@@ -273,3 +278,145 @@ def _sink(self, text, verbose):
+
+    Periodically freeze and unfreeze all embedding layers:
+
+    >>> def my_sched(net, _at, _fn):
+    ...    if len(net.history) % 2 == 0:
+    ...        return skorch.utils.freeze_parameter
+    ...    else:
+    ...        return skorch.utils.unfreeze_parameter
+    >>> cb = ParamMapper('embedding*.weight', schedule=my_sched)
+    >>> net = Net(myModule, callbacks=[cb])
+
+    Parameters
+    ----------
+    patterns : str or callable or list
+      The pattern(s) to match parameter names against. Patterns are
+      UNIX globbing patterns as understood by :func:`~fnmatch.fnmatch`.
+      Patterns can also be callables which will get called with the
+      parameter name and are regarded as a match when the callable
+      returns a truthy value.
+
+      Example: ``'linear*.weight'`` or ``['linear0.*', 'linear1.bias']``
+            or ``lambda name: name.startswith('linear')``.
+
+    fn : function
+      The function to apply to each parameter separately.
+
+    at : int or callable
+      In case you specify an integer it represents the epoch number the
+      function ``fn`` is applied to the parameters, in case ``at`` is
+      a function it will receive ``net`` as parameter and the function
+      is applied to the parameter once ``at`` returns ``True``.
+
+    schedule : callable or None
+      If specified it determines how ``fn`` and ``at`` are used.
+      This function receives ``net``, ``at`` and ``fn`` and returns
+      the function to be applied on the matched parameters.
+
+    """
+    def __init__(self, patterns, fn=noop, at=1, schedule=None):
+        self.at = at
+        self.fn = fn
+        self.schedule = schedule or self._default_schedule
+        self.patterns = patterns
+
+    def initialize(self):
+        if not isinstance(self.patterns, (list, tuple)):
+            self.patterns = [self.patterns]
+
+        if isinstance(self.at, int):
+            self.at = partial(self._epoch_at, epoch=self.at)
+
```

`initialize` should return `self`.

Also, I would like to see a check on `at` if it is `int`, because I can easily see someone setting `at=0`, which will just silently fail.